contentcuration/contentcuration/frontend/channelEdit/components/AssessmentEditor/AssessmentEditor.vue

```diff
@@ -352,11 +352,11 @@
       swapItems(firstItem, secondItem) {
         const firstUpdatedItem = {
           ...firstItem,
-          order: secondItem.order,
+          order: this.itemIdx(secondItem),
```

Comment:
I am not sure if I follow this update. `order` usually starts with 1 for most of our backend data, whereas indices are zero-based, so I am a bit afraid of running into edge-case situations when mixing it in this way. Have you located what was the reason for the order being set incorrectly, and is this change necessary to fix it? I am wondering whether the issue you fixed on line 375 might be sufficient.

Reply:
Ah, that's an important distinction, I didn't realize this!  I think you're right in that change on line 375 should be sufficient for now... The reason I made this change is because with the index from the `sortedItems`, it's guaranteed that no two items will ever have the same index, and these indices will also match the order that the user sees on the screen.  However, if it somehow works out that two items have the same `order` to start with, then performing the `swapItems` function will fail.

That said, I don't think this is the right solution for those problems, and could potentially cause more issues.

One idea I played with was to actually update the orders on all of the items every time two items swap, and to use the `itemIndex` function as the source.   That feels a little more solid to me, since it guarantees that even if two items have the same `order` property, that won't ripple out and cause other strange permutations and wrinkles to happen...  it will be "self-healing".